plugins/scale/scale.cpp

```diff
@@ -820,6 +872,63 @@ class wayfire_scale : public wf::plugin_interface_t
             return;
         }
 
+        std::vector<wayfire_view> filtered_views;
+        bool need_new_focus = false;
+        auto it = std::remove_if(views.begin(), views.end(),
+            [&filtered_views, &need_new_focus, this] (wayfire_view v)
+        {
+            scale_filter_signal signal{v, false};
+            output->emit_signal("scale-filter", &signal);
+            if (signal.hide)
+            {
+                filtered_views.push_back(v);
+                if (v == current_focus_view)
+                {
+                    need_new_focus = true;
+                }
+
+                return true;
+            }
+
+            return false;
+        });
+        views.erase(it, views.end());
+
+        if (need_new_focus || !current_focus_view)
+        {
+            current_focus_view = views.empty() ? nullptr : views.front();
+            output->focus_view(current_focus_view, true);
+        }
+
+        /* update hidden views */
+        for (auto v : filtered_views)
+        {
+            /* ensure that this view has a transformer and is in scale_data */
+            add_transformer(v);
+            auto& view_data = scale_data[v];
+            if (view_data.hidden == view_scale_data::hidden_t::SHOWN)
+            {
+                view_data.hidden = view_scale_data::hidden_t::HIDING;
+                setup_view_transform(view_data, 1, 1, 0, 0, 0);
+            }
+
+            for (auto& child : v->enumerate_views(false))
+            {
+                if (child == v)
```

Comment:
why do we need special handling for child and view when they seem to have the exact same transform applied to them?

Reply:
You're right, there is not need for this.